src/battle_regulation.c

```diff
@@ -0,0 +1,276 @@
+#include "battle_regulation.h"
+
+#include "global.h"
+
+#include "constants/items.h"
+
+#include "dex_mon_measures.h"
+#include "party.h"
+#include "pokemon.h"
+#include "unk_020290B4.h"
+
+static BOOL sub_02074C50(u16 *a0, u16 *a1, u16 *a2, int a3, int a4, int a5, int a6);
+
+PokedexData *sub_02074944(HeapID heapId) {
```

Comment:
Can be named to something like Pokedex_CreateAndLoad, Pokedex_Load, etc, whatever you see fit